### PR TITLE
chore: sync app with latest unreleased sdk + bump sdk version

### PR DESCRIPTION
**Pull request by gminn**
### Summary

Sync this app with latest (unreleased) SDK. Needed for internal CI
to pass. Additionally, the SDK has been bumped to latest (1.15.0).

Confirmed only expected differences between this app and the
current unreleased sdk:

<details>

```
Only in .: .git
Only in .: .github
diff --color -r /Users/gminnehan/memfault/sdk/embedded/examples/esp32/apps/memfault_demo_app/.gitignore ./.gitignore
0a1
> /build
Only in .: .gitmodules
diff --color -r /Users/gminnehan/memfault/sdk/embedded/examples/esp32/apps/memfault_demo_app/CMakeLists.txt ./CMakeLists.txt
17a18,21
> # Look for the Memfault SDK in a subdirectory first, when this app is used
> # standalone (not from within the Memfault SDK)
> get_filename_component(memfault_firmware_sdk_dir third-party/memfault-firmware-sdk ABSOLUTE)
> if(NOT EXISTS ${memfault_firmware_sdk_dir})
19,36d22
< # If we found the Memfault SDK, include it in the build
< if(EXISTS ${memfault_firmware_sdk_dir}/ports/esp_idf/memfault.cmake)
<   include(${memfault_firmware_sdk_dir}/ports/esp_idf/memfault.cmake)
< else()
<   # Otherwise, append this to the main/idf_component.yml file:
<   #   memfault/memfault-firmware-sdk: "*"
<   # Ideally we'd push an environment variable and use a conditional dependency
<   # https://docs.espressif.com/projects/idf-component-manager/en/latest/reference/manifest_file.html#conditional-dependencies
<   # But that requires idf-component-manager v2+, which is not available on all
<   # esp-idf versions
<
<   # if the string "memfault/memfault-firmware-sdk" isn't in the file, append it:
<   file(READ ${CMAKE_CURRENT_SOURCE_DIR}/main/idf_component.yml idf_component_yml)
<   if(NOT idf_component_yml MATCHES "memfault/memfault-firmware-sdk")
<     file(APPEND ${CMAKE_CURRENT_SOURCE_DIR}/main/idf_component.yml
<       "  memfault/memfault-firmware-sdk: \"*\"\n"
<     )
<   endif()
37a24
> include(${memfault_firmware_sdk_dir}/ports/esp_idf/memfault.cmake)
Only in .: sdkconfig
Only in .: third-party
```
</details>

### Test Plan

- CI
- Flashed an esp32c3 devkitm

### CODE REVIEW

**Comment by gminn** [comment]
* **#2** <a href="https://app.graphite.dev/github/pr/memfault/esp32-standalone-example/2?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* `main`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @gminn and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: CMakeLists.txt [hunk 1]
```diff
@@ -15,13 +15,26 @@ if(DEFINED IDF_VERSION_MAJOR)
   endif()
 endif()
 
-# Look for the Memfault SDK in a subdirectory first, when this app is used
```
Comment (noahp) [inline CMakeLists.txt]: We need this for git submodule method

File: main/memfault/memfault_metrics_heartbeat_config.def
Comment (ejohnso49) [inline main/memfault/memfault_metrics_heartbeat_config.def:1]: These are just duplicates right?

**Review by ejohnso49 — APPROVED** [review]

File: CMakeLists.txt (hunk 1 above)
Comment (gminn) [reply CMakeLists.txt]: Ah shoot ofc, will add back!

File: main/memfault/memfault_metrics_heartbeat_config.def
Comment (gminn) [reply main/memfault/memfault_metrics_heartbeat_config.def:1]: Yep!

**Pull request merged**